compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -75,11 +76,27 @@
 
     private ClassLoader classLoader;
     private SymbolTable symbolTable;
+    private final BType[] definedReadOnlyMemberTypes;
 
     JMethodResolver(ClassLoader classLoader, SymbolTable symbolTable) {
 
         this.classLoader = classLoader;
         this.symbolTable = symbolTable;
+        this.definedReadOnlyMemberTypes = new BType[]{
```

Comment:
then, can this be moved to symbolTable?

Reply:
We don't need this collection of types anywhere else since the read-only analysis is generally based on a flag + helper methods in `Types.java`. But for the method resolver we need these specifically for an initial `isValidReturnBType` check. 

Shall we leave this here since it may not make sense in other contexts?

Reply:
ok